src/components/shared/Entity/EndpointConfig/index.tsx

```diff
@@ -99,37 +98,64 @@ function EndpointConfig({
 
     // Storing flag to handle knowing if a config changed
     //  during both create or edit.
-    const resetEndpointConfig = useMemo(
-        () =>
-            // TODO (connectors) We need to break this single flag into at least two
-            //  1 - if we want to reset data (only should happen on create when connector changes)
-            //  2 - if we need to update the schema (can happen on create and in edit)
-            editWorkflow && !unsupportedConnectorVersion
-                ? false
-                : connectorTag?.endpoint_spec_schema &&
-                  !isEqual(connectorTag.endpoint_spec_schema, endpointSchema),
-        [
-            connectorTag?.endpoint_spec_schema,
-            editWorkflow,
-            endpointSchema,
-            unsupportedConnectorVersion,
-        ]
-    );
+    const [resetEndpointConfig, updateSchema] = useMemo(() => {
+        let resetConfigResponse = false;
+        let updateSchemaResponse = false;
+
+        const schemaChanged = Boolean(
+            connectorTag?.endpoint_spec_schema &&
+                !isEqual(connectorTag.endpoint_spec_schema, endpointSchema)
+        );
+
+        if (editWorkflow) {
+            // In edit we never want to clear the config since any changes to
+            //  the schema should come from an upgrade and connectors should
+            //  be backwards compatible anyway (as of Q1 2024)
+            resetConfigResponse = false;
+
+            // We do want to reset the schema if it is known to be unsupported and
+            //   the schema has changed
+            updateSchemaResponse = unsupportedConnectorVersion && schemaChanged;
+        } else {
+            // In create if the schema changed it probably means the user selected
+            //  a different connector in the dropdown. So we need to clear out data
```

Comment:
> So we need to clear out data

The use of _data_ is similarly a little ambiguous here; I would specify that the _data_ being cleared is the endpoint configuration.